### PR TITLE
DUPLO-16119 Add name to duplocloud_aws_cloudfront_distribution

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Added name attribute, made comment attribute depreciated
## Summary of changes

This PR does the following:

- added name attribute in duplocloud_aws_cloudfront_distribution resource schema
- made comment attribute depreceated

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-16119 [TF ] Add name to duplocloud_aws_cloudfront_distribution](https://app.clickup.com/t/8655600/DUPLO-16119)

File: docs/resources/aws_cloudfront_distribution.md [hunk 1]
```diff
@@ -103,14 +103,15 @@ resource "duplocloud_aws_cloudfront_distribution" "cfd" {
...
+- `name` (String) Name of the distributiion
 - `origin` (Block Set, Min: 1) (see [below for nested schema](#nestedblock--origin))
 - `tenant_id` (String) The GUID of the tenant that the aws cloudfront distribution will be created in.
 - `viewer_certificate` (Block List, Min: 1, Max: 1) (see [below for nested schema](#nestedblock--viewer_certificate))
 
 ### Optional
 
 - `aliases` (Set of String) Extra CNAMEs (alternate domain names), if any, for this distribution.
-- `comment` (String) Any comments you want to include about the distribution.
+- `comment` (String, Deprecated) Any comments you want to include about the distribution. comment has been depreciated instead use name
```
Comment (zafarabbas) [inline docs/resources/aws_cloudfront_distribution.md:114]: typo: `deprecated`

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go [hunk 2]
```diff
@@ -49,6 +49,13 @@ func duploAwsCloudfrontDistributionSchema() map[string]*schema.Schema {
 			Optional:     true,
 			Computed:     true,
 			ValidateFunc: validation.StringLenBetween(0, 128),
+			Deprecated:   "comment has been depreciated instead use name",
```
Comment (zafarabbas) [inline duplocloud/resource_duplo_aws_cloudfront_distribution.go:52]: typo: `deprecated`

**Review by zafarabbas — CHANGES_REQUESTED** [review]

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go [hunk 3]
```diff
@@ -49,6 +49,13 @@ func duploAwsCloudfrontDistributionSchema() map[string]*schema.Schema {
...
+		},
+		"name": {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_aws_cloudfront_distribution.go:54]: We should not add the required attribute `name`, We are forcing existing customer code to be modified on provider upgrade. Instead we can add optional name attribute  `ConflictsWith` comment attribute.

**Comment by Duplo-nikv** [comment]
Ticket mentioned to make it required

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go (hunk 3 above)
Comment (zafarabbas) [reply duplocloud/resource_duplo_aws_cloudfront_distribution.go:54]: @tahir-duplo should we even deprecate `comment` in this version?

Comment (tahir-duplo) [reply duplocloud/resource_duplo_aws_cloudfront_distribution.go:54]: Yes, We should deprecate `comment`

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go [hunk 4]
```diff
@@ -1032,7 +1042,7 @@ func expandAwsCloudfrontDistributionConfig(d *schema.ResourceData) *duplosdk.Dup
 		PriceClass:           &duplosdk.DuploStringValue{Value: d.Get("price_class").(string)},
 		WebACLId:             d.Get("web_acl_id").(string),
 	}
-
+	distributionConfig.Comment = d.Get("name").(string)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_aws_cloudfront_distribution.go:1045]: @Duplo-nikv We should add a check if the name is provided use name or a comment.

**Review by zafarabbas — APPROVED** [review]

**Comment by zafarabbas** [comment]
@tahir-duplo can you review again?

File: duplocloud/resource_duplo_aws_cloudfront_distribution.go [hunk 5]
```diff
@@ -1609,6 +1624,7 @@ func flattenAwsCloudfrontDistribution(d *schema.ResourceData, duplo *duplosdk.Du
 
 	if duplo.Comment != "" {
 		d.Set("comment", duplo.Comment)
+		d.Set("name", duplo.Comment)
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_aws_cloudfront_distribution.go:1627]: We should either set `comment` or `name`.

**Pull request merged**